src/main/java/org/jenkinsci/plugins/github_branch_source/BaseGitHubWebHookUtil.java

```diff
@@ -0,0 +1,347 @@
+            String code = err.getString("code");
+            throw new RuntimeException(code);
+        }
+        else {
+            String apiKey = res.getString("token");
+
+            // Would use the GitHubTokenCredentialsCreator, but the credential descriptions are not good
+            // and it creates incompatible credential types
+            return createCredentials(apiUrl, username, apiKey, uuid.substring(uuid.length()-4));
+        }
+    }
+
+    /**
+     * Create a github usesrname/password credential with the apiKey; must match what the scan lookup needs
+     */
+    private UsernamePasswordCredentialsImpl createCredentials(String apiUrl, String username, String apiKey, String identifier) {
+        String description = String.format("%s GitHub auth token *%s", username, identifier);
+        final UsernamePasswordCredentialsImpl creds = new UsernamePasswordCredentialsImpl(
+                CredentialsScope.GLOBAL,
+                UUID.randomUUID().toString(),
+                description,
+                username,
+                apiKey);
+
+        URI serverUri = URI.create(apiUrl = StringUtils.defaultIfBlank(apiUrl, Connector.GITHUB_CREDENTIAL_DOMAIN_URI));
+
+        List<DomainSpecification> specifications = Arrays.asList(
+            new SchemeSpecification(serverUri.getScheme()),
+            new HostnameSpecification(serverUri.getHost(), null)
+        );
+
+        final Domain domain = new Domain(serverUri.getHost(), serverUri.getHost(), specifications);
+        ACL.impersonate(ACL.SYSTEM, new Runnable() { // do it with system rights
+            @Override
+            public void run() {
+                try {
+                    new SystemCredentialsProvider.StoreImpl().addDomain(domain, creds);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+
+        return creds;
+    }
+
+    /* (non-Javadoc)
+     * @see org.jenkinsci.plugins.github_branch_source.GitHubWebHookUtil#registerOrganizationWebhook(jenkins.scm.api.SCMSourceOwner, java.lang.String, java.lang.String, java.lang.String)
+     */
+    @Override
+    public void registerOrganizationWebhook(SCMSourceOwner scmSource, String apiUrl, String repoOwner, String scanCredentialsId) throws Exception {
+        StandardCredentials credentials = Connector.lookupScanCredentials(scmSource, apiUrl, scanCredentialsId);
+        UsernamePasswordCredentials creds = (UsernamePasswordCredentials)credentials;
+
+        String jenkinsUrl = Jenkins.getActiveInstance().getRootUrl();
+
+        if (!isValidExternalJenkinsUrl(jenkinsUrl)) {
+            logger.warning(String.format("Skipping webhook registration for invalid external URL: %s", jenkinsUrl));
+            return;
+        }
+
+        if (!jenkinsUrl.endsWith("/")) {
+            jenkinsUrl += "/";
+        }
+        String jenkinsCallbackUrl = jenkinsUrl + "github-webhook/";
```

Comment:
[`GithubWebHook.URLNAME`](https://github.com/jenkinsci/github-plugin/blob/8630c49e7c9821cf6bcc3ad1e0dfb79cbb7936a6/src/main/java/com/cloudbees/jenkins/GitHubWebHook.java#L44) should be used.